experiment/src/org/labkey/experiment/samples/SampleTypeAndDataClassFolderImporter.java

```diff
@@ -143,7 +146,11 @@ else if (file.startsWith(SampleTypeAndDataClassFolderWriter.DATA_CLASS_PREFIX))
                     // handle wiring up any derivation runs
                     if (runsXarFile != null)
                     {
-                        XarSource runsXarSource = new CompressedInputStreamXarSource(xarDir.getInputStream(runsXarFile.getFileName().toString()), runsXarFile, logFile, job, ctx.getUser(), ctx.getContainer(), ctx.getXarJobIdContext());
+                        XarSource runsXarSource;
+                        if (runsXarFile.getFileName().toString().endsWith(".xar.xml"))
```

Comment:
```suggestion
                        if (runsXarFile.getFileName().toString().toLowerCase().endsWith(".xar.xml"))
```

Might want to toss a .toLowerCase() in here, though I think we should always be creating files with lower case extensions. FolderXarImporterFactory.getXarSource() probably should do that too.